modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp

```diff
@@ -380,78 +380,76 @@ renderResponse.setTitle(headerTitle);
 		}
 	}
 
-	Liferay.provide(
-		window,
-		'<portlet:namespace />selectFileEntryType',
-		function (fileEntryTypeId, fileEntryTypeName) {
-			var A = AUI();
+	window['<portlet:namespace />selectFileEntryType'] = function (
+		fileEntryTypeId,
+		fileEntryTypeName
+	) {
+		var A = AUI();
 
-			var searchContainer = Liferay.SearchContainer.get(
-				'<portlet:namespace />dlFileEntryTypesSearchContainer'
-			);
+		var searchContainer = Liferay.SearchContainer.get(
```

Comment:
I think we went through this in previous PRs... now you depend on `liferay-search-container` here implicitly and there's no way to know when this will work and when it won't.

Reply:
Yes, we did, but I still wanted to send this PR to talk about it:

I added the `use="liferay-search-container"` attribute to the `<aui:script>` tag [here](https://github.com/liferay-frontend/liferay-portal/blob/b60b4562efce413acaa6bf0c321e0620b8d1df69/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp#L333).

After deploying that, the call to the `savePage` function made [here](https://github.com/liferay-frontend/liferay-portal/blob/b60b4562efce413acaa6bf0c321e0620b8d1df69/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp#L82) failed saying that `<portlet:namespace />savePage` function was undefined. 
(The `savePage` function is declared [here](https://github.com/liferay-frontend/liferay-portal/blob/b60b4562efce413acaa6bf0c321e0620b8d1df69/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp#L366))

Reply:
When you add `use` to `aui:script`, it will create a closure that will hide `function savePage` within it instead of pushing it to the global scope as it is without it.

Reply:
oh then, that makes sense - didn't see it
I'll fix that